File: examples/io_grpc/bazel_build_java/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/services/HelloWorldServiceAdapter.java
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.services;

import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorld;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorldServiceGrpc;
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorldServiceGrpc.AsyncService;
import io.grpc.BindableService;
import io.grpc.ServerServiceDefinition;
import io.grpc.stub.StreamObserver;
import java.util.concurrent.Callable;
import javax.annotation.processing.Generated;

@Generated("to be generated")
public class HelloWorldServiceAdapter implements BindableService, AsyncService {
  private final Callable<AsyncService> service;

  public HelloWorldServiceAdapter(Callable<AsyncService> service) {
    this.service = service;
  }

  @Override
  public void sayHello(
      HelloWorld.SayHelloRequest req, StreamObserver<HelloWorld.SayHelloResponse> resp) {
    try {
      service.call().sayHello(req, resp);
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }

  @Override
  public void sayGoodbye(
      HelloWorld.SayGoodbyeRequest req, StreamObserver<HelloWorld.SayGoodbyeResponse> resp) {
    try {
      service.call().sayGoodbye(req, resp);
    } catch (Exception e) {
      throw new RuntimeException(e);
    }
  }

  @Override
  public ServerServiceDefinition bindService() {
    return HelloWorldServiceGrpc.bindService(this);
  }
}

File: examples/io_grpc/bazel_build_kt/service/armeria/src/main/kotlin/com/geekinasuit/daggergrpc/iogrpc/example/armeria/services/HelloWorldServiceAdapter.kt
package com.geekinasuit.daggergrpc.iogrpc.example.armeria.services

import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorld.SayGoodbyeRequest
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorld.SayGoodbyeResponse
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorld.SayHelloRequest
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorld.SayHelloResponse
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorldServiceGrpc
import com.geekinasuit.daggergrpc.iogrpc.example.armeria.api.HelloWorldServiceGrpc.AsyncService
import io.grpc.BindableService
import io.grpc.ServerServiceDefinition
import io.grpc.stub.StreamObserver
import javax.annotation.Generated

@Generated("to be generated")
class HelloWorldServiceAdapter(val service: () -> AsyncService) : BindableService, AsyncService {
  override fun sayHello(req: SayHelloRequest, resp: StreamObserver<SayHelloResponse>) =
    service().sayHello(req, resp)

  override fun sayGoodbye(req: SayGoodbyeRequest, resp: StreamObserver<SayGoodbyeResponse>) =
    service().sayGoodbye(req, resp)

  override fun bindService(): ServerServiceDefinition = HelloWorldServiceGrpc.bindService(this)
}
